bon-macros/src/builder/builder_gen/setters/mod.rs

```diff
@@ -166,6 +179,29 @@ impl<'a> SettersCtx<'a> {
         Ok([self.setter_method(some_fn), self.setter_method(option_fn)].concat())
     }
 
+    fn getter_method(&self, item: Item) -> TokenStream {
+        let Item { name, vis, docs } = item;
+
+        let index = &self.member.index;
+        let state_var = &self.base.state_var;
+        let member_pascal = &self.member.name.pascal;
+        let normalized_type = &self.member.underlying_norm_ty();
+        let state_mod = &self.base.state_mod.ident;
+
+        quote! {
+            #( #docs )*
+            #[allow(clippy::inline_always)]
+            #[inline(always)]
+            #vis fn #name(&self) -> &#normalized_type
```

Comment:
We can't guarantee that optional members are set with `Some`, so the current impl is unsound.

For members of the special type `Option<T>` the `underlying_norm_ty()` returns `T`. Instead, for `!member.is_required()` this method needs to return `Option<&T>` and it doesn't need an `unwrap_unchecked` in that case.